app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/banglejs/BangleJSCoordinator.java

```diff
@@ -0,0 +1,150 @@
+package nodomain.freeyourgadget.gadgetbridge.devices.banglejs;
+
+import java.util.Collection;
+import java.util.Collections;
+
+
+import android.app.Activity;
+import android.content.Context;
+import android.net.Uri;
+import android.bluetooth.le.ScanFilter;
+import android.os.ParcelUuid;
+import androidx.annotation.NonNull;
+
+import nodomain.freeyourgadget.gadgetbridge.GBException;
+import nodomain.freeyourgadget.gadgetbridge.devices.AbstractDeviceCoordinator;
+import nodomain.freeyourgadget.gadgetbridge.devices.InstallHandler;
+import nodomain.freeyourgadget.gadgetbridge.devices.SampleProvider;
+import nodomain.freeyourgadget.gadgetbridge.devices.hplus.HPlusConstants;
+import nodomain.freeyourgadget.gadgetbridge.devices.id115.ID115SampleProvider;
+import nodomain.freeyourgadget.gadgetbridge.entities.DaoSession;
+import nodomain.freeyourgadget.gadgetbridge.entities.Device;
+import nodomain.freeyourgadget.gadgetbridge.entities.HPlusHealthActivitySampleDao;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDevice;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDeviceCandidate;
+import nodomain.freeyourgadget.gadgetbridge.model.ActivitySample;
+import nodomain.freeyourgadget.gadgetbridge.model.DeviceType;
+
+public class BangleJSCoordinator extends AbstractDeviceCoordinator {
+
+    @Override
+    public DeviceType getDeviceType() {
+        return DeviceType.BANGLEJS;
+    }
+
+    @Override
+    public String getManufacturer() {
+        return "Espruino";
+    }
+
+    @Override
+    public Collection<? extends ScanFilter> createBLEScanFilters() {
+        // TODO: filter on name beginning Bangle.js? Doesn't appear to be built-in :(
+        // https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder.html#setDeviceName(java.lang.String)
+        ParcelUuid hpService = new ParcelUuid(BangleJSConstants.UUID_SERVICE_NORDIC_UART);
+        ScanFilter filter = new ScanFilter.Builder().setServiceUuid(hpService).build();
+        return Collections.singletonList(filter);
+    }
+
+    @Override
+    public DeviceType getSupportedType(GBDeviceCandidate candidate) {
+        String name = candidate.getDevice().getName();
+        /* Filter by Espruino devices to avoid getting
+        the device chooser full of spam devices. */
+        if (name != null && (
+              name.startsWith("Bangle.js") ||
+              name.startsWith("Pixl.js") ||
+              name.startsWith("Puck.js") ||
+              name.startsWith("MDBT42Q") ||
+              name.startsWith("Espruino"))) 
+            return DeviceType.BANGLEJS;
+
+        return DeviceType.UNKNOWN;
+    }
+
+    @Override
+    public int getBondingStyle(){
+        return BONDING_STYLE_NONE;
```

Comment:
BONDING_STYLE_NONE means that the device should not be paired the bluetooth level for BLE devices chose to do some custom key exchange on a higher layer. Did you have that in mind?

Reply:
I assumed that it was such that if Bangle.js specifically requested bonding you'd have it, but if not then you wouldn't? As Bangle.js does have the ability to do bonding I think you're right, BONDING_STYLE_ASK would probably be preferable.

Reply:
BONDING_STYLE_ASK is currently not used by any device AFAIK, hope it works ;)

So bangle.js has zero security and anyone could connect?

Reply:
Ahh, sorry. Seems to work though. I saw the comments and they said it was preferable to use BONDING_STYLE_ASK :)

By default there's zero security, yes. There's a bunch of stuff it can do (including whitelisting) but it's not enabled by default now. At the start it used to be, but I was getting so many support requests like "I paired this with A and now it's broken" that I had to disable it :)